scriptler/discardOldBuilds.groovy

```diff
@@ -15,12 +15,20 @@
 // artifactDaysToKeep: If not -1 nor null, artifacts are only kept up to this days.
 // artifactNumToKeep: If not -1 nor null, only this number of builds have their artifacts kept.
 
-noLogRotation = hudson.model.Hudson.instance.items.findAll
-{job -> job.isBuildable() && job.logRotator==null}
-noLogRotation.each() { job ->
-    println job.name
-    if(!"true".equals(dryRun)){
-        job.logRotator = new hudson.tasks.LogRotator ( daysToKeep, numToKeep, artifactDaysToKeep, artifactNumToKeep)
-        println "$job.name fixed "
+import jenkins.model.Jenkins
+import hudson.tasks.LogRotator
+
+def setLogRotator(def instance) {
+    instance.items.findAll {job -> job.class.simpleName == 'Folder'}.each { setLogRotator(it) }
+    noLogRotation = instance.items.findAll {job -> job.isBuildable() && job.logRotator==null}
```

Comment:
This is gonna break recursively if _job_ is a _Folder_ because _isBuildable()_ is only applicable to _Job_.

Overall, the recursion may not be necessary here. The Jenkins API provides a way to get all item of a certain Class in your entire instance with something like:

`Jenkins.instance.getAllItems(Job.class)`